statics/js/components/rule-detail.js

```diff
@@ -788,41 +754,25 @@ Vue.component('rule-detail', {
     getRules: function() {
       var self = this;
       console.log(this.filters);
-      var queryBridge = "G.V('" + self.bridge.id + "').As('bridge')";
-      var queryPorts = queryBridge + ".Out().Has('Type', 'ovsport').As('ovsports').Out().As('ovsinterfaces')";
-      var queryRules = queryBridge + ".Out().Has('Type', 'ofrule')";
-      var has = "";
-      var list = [];
-      var i = 0;
-      for (var k in this.filters) {
-        has += "'filters', regex('.*" + k + "=" + this.filters[k] + ".*')";
-        list[i] = k;
-        i +=1;
-      }
-      var query = queryBridge + "." + queryPorts + "." ;
-      if (has.length > 0) {
-        for (var k in this.filters) {
-          query += queryRules + ".Has('filters', regex('.*" + k + "=" + this.filters[k] + ".*')).As('" + k + "').";
-        }
-      }
-      else{
-        query += queryRules + ".As('ofrules')."
-      }
-      query += "Select('bridge', 'ovsports', 'ovsinterfaces'";
-      if (has.length > 0){
-        for (var p in this.filters){
-          query += ", '" + p + "'";
-        }
+      var keys = Object.keys(this.filters);
+      if(keys.length == 0) {
+        // Short cut no filter.
+        this.filtered = null;
+        this.memoBridgeLayout = null;
+        return;
       }
-      else{
-        query += ", 'ofrules'";
+      var queryRules = "G.V('" + self.bridge.id + "').Out().Has('Type', 'ofrule')";
```

Comment:
@safchain 
Hello, I do like this "gremlin query builder" implementation.
Is it possible to use it here ? to build a gremlin query ? Right now this component responsible for so many things...
https://github.com/skydive-project/skydive/blob/master/js/api.ts#L675

Reply:
@sergeyglazyrindev-samsung It will be possible as it is part of the NPM module. I will push a draft of this package today as part of the skydive sources. But currently I think this implementation is OK. Just need to test it.